### PR TITLE
Add a search engine for core.ac.uk

### PR DESCRIPTION
**Pull request by return42**
## What does this PR do?

Add a search engine for core.ac.uk : https://core.ac.uk/

Initial commit b416231 was fetched-from: https://github.com/searx/searx/pull/2688 / for  the rest see commit messages.

### CODE REVIEW

File: searx/engines/core.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+from urllib.parse import urlencode
+
+from searx import logger
+from searx.exceptions import SearxEngineAPIException
+
+logger = logger.getChild('CORE engine')
+
+about = {
+    "website": 'https://core.ac.uk',
+    "wikidata_id": None,
```
Comment (dalf) [inline searx/engines/core.py]: The wikidata ID is Q22661180

Comment (return42) [reply searx/engines/core.py]: Thanks!  .. done.

File: searx/engines/core.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    "use_official_api": True,
+    "require_api_key": True,
+    "results": 'JSON',
+}
+
+categories = ['science']
+paging = True
+nb_per_page = 10
+
+api_key = 'unset'
```
Comment (dalf) [inline searx/engines/core.py:29]: If a key is required, the engine definition in settings.yml should be in comments

Comment (return42) [reply searx/engines/core.py:29]: Ooops .. I forgot .. done!

File: searx/settings.yml [hunk 3]
```diff
@@ -273,6 +273,15 @@ engines:
...
 
+  - name : core.ac.uk
+    engine : core
+    categories : science
+    shortcut : cor
+    # get your API key from: https://core.ac.uk/api-keys/register/
+    # api_key : "xxxxxxxx"
+    # set api_key and comment out disabled ..
+    disabled: True
+
```
Comment (dalf) [inline searx/settings.yml:281]: It is the whole engine definition that should be comment out : without the api_key nothing works AFIK.

**Pull request merged**